dingo/gw/waveform_generator/waveform_generator.py

```diff
@@ -776,10 +769,25 @@ class NewInterfaceWaveformGenerator(WaveformGenerator):
     """Generate polarizations using GWSignal routines in the specified domain for a
     single GW coalescence given a set of waveform parameters.
     """
-
     def __init__(self, **kwargs):
         WaveformGenerator.__init__(self, **kwargs)
 
+        try:
+            # we want to import the new interface, but we don't want to import it 
+            # for all users because just importing the module causes unintended side effects
+            # for example, you cannot pass the debugger through the import statement.
+            # Thus we only import it when the class is instantiated.
+            # However, we don't want to reimport the module every time we need to 
+            # use the function as this is costly. Therefore, we import it once 
+            # when the class is instantiated and store it in the global namespace
+            from lalsimulation.gwsignal.core import waveform
+            from lalsimulation.gwsignal.models import gwsignal_get_waveform_generator
+            globals()['gws_wfm'] = waveform
+            globals()["new_interface_get_waveform_generator"] = gwsignal_get_waveform_generator 
+        except ImportError:
+            raise ImportError("The new interface is not available, please install lalsimulation")
```

Comment:
Shouldn't this say to install `pyseobnr`, not `lalsimulation`?

We should make this error message a bit more detailed.